planner/min_priority_queue.go

```diff
@@ -1,27 +1,29 @@
 package planner
 
-type MinPriorityQueue []Vertex
+type MinPriorityQueue[T PriorityQueueItem] struct {
```

Comment:
Oh... this is actually quite similar to template class in C++, which allows generic datatypes :)